qiskit_braket_provider/providers/adapter.py

```diff
@@ -427,10 +430,16 @@ def to_braket(
     basis_gates = set(basis_gates or _TRANSLATABLE_QISKIT_GATE_NAMES)
 
     braket_circuit = Circuit()
-    if not verbatim and not {gate.name for gate, _, _ in circuit.data}.issubset(
-        basis_gates
+    if not verbatim and (
+        not {gate.name for gate, _, _ in circuit.data}.issubset(basis_gates)
+        or connectivity
     ):
```

Comment:
Consider refactoring for readability, perhaps:
```suggestion
    uses_only_basis_gates = {gate.name for gate, _, _ in circuit.data}.issubset(basis_gates)
    needs_transpilation = connectivity or not uses_only_basis_gates
    if not verbatim and needs_transpilation:
```